linux-%.bbappend: Add IMA patch fixing unneeded re-appraisal

rojkov opened the pull request:
The patch taken from https://sourceforge.net/p/linux-ima/mailman/message/34894743/
makes IMA not to re-appraise a new file if only its timestamps
have changed during extraction from a tarball.

This is needed to avoid resetting the security.ima attribute for files
extracted from tarballs with software updates.

Signed-off-by: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>

pohly commented on meta-integrity/recipes-kernel/linux/linux/0001-ima-fix-ima_inode_post_setattr.patch:
```diff
@@ -0,0 +1,35 @@
+
+Reported-by: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>
+Signed-off-by: Mimi Zohar <zohar@linux.vnet.ibm.com>
+---
```
Upststream-Status: Submitted [link to IMA mailing list post] needs to be added. If it's already in some kernel tree, even better.

rojkov commented:
Added Upstream-Status tag and force pushed. No patch in Mimi's repo [1] so far.

[1] https://git.kernel.org/cgit/linux/kernel/git/zohar/linux-integrity.git/

pohly commented:
@rojkov which kernel did you test this patch with? I'm having problems getting it applied to both 4.4.3 and 4.1.18 (MACHINE=qemu-x86), patch needs to be refreshed.

rojkov commented:
@pohly I did that with 4.1.15+git0+46bb64d605_788dfc9859 (intel-corei7-64)

pohly commented:
@rojkov could there have been some copy-and-paste which broke the patch because it changed tabs into spaces or vice versa? After re-generating the patch I got a slightly different one (see https://github.com/01org/meta-intel-iot-security/commit/ba025e9289683d0cc8e19639bd6cb49ee667699d).

rojkov commented:
@pohly Yeah, I guess I've finally figured out what's wrong with the tabs: my git removes leading whitespaces automatically from patch files I add to a repo even though `git format-patch` produces correct patches with leading whitespaces in context lines. And they seem to be required by older version of `patch`.

I'll re-add the patch and force push.

pohly commented:
@rojkov can you fetch https://github.com/01org/meta-intel-iot-security/commit/ba025e9289683d0cc8e19639bd6cb49ee667699d and squash that into your commit?

Besides fixing the patch, I have also added a kernel version check. I want to have the ability to skip the patch when moving to a linux version which already includes it, without having to remove the patch entirely.

I'm also not sure whether the patch works with older OE releases, so I'm simply skipping it there.

pohly commented:
And please also rebase to current master.

rojkov commented:
Sure, after several attempts I managed to --ignore-whitespace...

1. re-added the patch with leading whitespaces in the context lines
2. squashed in https://github.com/01org/meta-intel-iot-security/commit/ba025e9289683d0cc8e19639bd6cb49ee667699d
3. rebased
4. force pushed

pohly commented:
All tests clean now, merging.
